### ISSUE TITLE
[TorchToLinalg] Add support for AtenReplicationPad1dOp and lower to linalg backend

### ISSUE THREAD

**Issue by zahidwx** [llvm/torch-mlir#4216]
### Problem

- `AtenReplicationPad1dOp` is not available in torch dialect. There is no lowering to linalg backend.
https://docs.pytorch.org/docs/stable/generated/torch.nn.ReplicationPad1d.html
- Currently `AtenPadOp` decomposition in torch dialect uses replication pad 2d for cases which require the usage of 1d replication pad. Update the decomposition by making use of `AtenReplicationPad1dOp` for 1d padding with "**replicate**" mode.

**Comment by zahidwx** [comment]
Related PR: https://github.com/llvm/torch-mlir/pull/4217 merged to main.
